README.md

```diff
@@ -28,85 +29,90 @@ The following table contains nrich versions with corresponding minimum Java and
+Queries can be automatically formed from different types of inputs, such as a query class or a string with a list of search fields.
+Also, special configuration class is used to define how provided query values are used in query creation.
 
-### nrich-security-csrf
+### [nrich-security-csrf](nrich-security-csrf/README.md)
 
-Intended as a replacement for Spring Security csrf functionality. It works with Spring Web MVC and WebFlux libraries. Clients should define initial token url and after that send generated token with
-each request in a header or as a parameter.
+Intended as a replacement for Spring Security csrf functionality.
+`nrich-security-csrf` works with both Spring Web MVC and WebFlux libraries.
+Clients should define the initial token url and after that send the generated token with each request in a header or as a parameter.
 
-### nrich-spring-boot
+### [nrich-spring-boot](nrich-spring-boot/README.md)
 
-Adds additional functionality to Spring Boot. i.e. `ConditionalOnPropertyNotEmpty` annotation that checks if a property is not empty.
+Adds additional functionality to Spring Boot that is used within nrich.
 
-### nrich-spring
+### [nrich-spring](nrich-spring/README.md)
 
-Contains utility classes for Spring access. i.e. `ApplicationContextHolder` for resolving of `ApplicationContext`
-from static context.
+Contains utility classes for Spring access, such as `ApplicationContextHolder` for resolving of `ApplicationContext` from static context.
 
-### nrich-validation
+### [nrich-validation](nrich-validation/README.md)
 
-Contains additional `jakarta-validation-api` constraints and validators (i.e `ValidFile`, `MaxSizeInBytess` etc).
+Contains additional `jakarta-validation-api` constraints and validators that proved to be commonly used, such as `NotNullWhen` validator.
 
-### nrich-webmvc
+### [nrich-webmvc](nrich-webmvc/README.md)
 
-Provides additional functionality built on top of Spring Web MVC framework. Main purpose is handling exceptions through `NotificationErrorHandlingRestControllerAdvice` that handles exceptions by
-logging them and sending notifications to client with exception descriptions. It also adds additional classes that handle binding (i.e. transforming empty string to null), locale resolving etc.
+Provides additional functionality built on top of the Spring Web MVC framework.
+Main purpose is to handle exceptions through `NotificationErrorHandlingRestControllerAdvice` that handles exceptions by logging them,
+and then creating and sending notifications to the client-side with exception's description.
+The module uses the [nrich-notification](nrich-notification/README.md) module for notification handling and [nrich-logging](nrich-logging/README.md) for log handling.
+`nrich-webmvc` also adds additional classes that handle binding (in other words transforming empty string to null) and locale resolving.
```

Comment:
I prefer `i.e. transforming empty string to null` over `in other words transforming empty string to null`.
This version sounds like this is only functionality related to binding. I suppose that is not the case?

Reply:
I tried to lower the usage of i.e.by replacing them with semantically similar phrases because they are perhaps a bit easier to read, but in some cases i.e. actually fits better, such as this one.